packages/api/tests/endpoints/async-operations/test-create-async-operations.js

```diff
@@ -0,0 +1,480 @@
+    id: asyncOperation.id,
+  });
+
+  const pgRecord = await asyncOperationPgModel.search(
+    t.context.testKnex,
+    { id: asyncOperation.id }
+  );
+
+  const [asyncOperationPgRecord] = pgRecord;
+
+  t.true(fetchedRecord.createdAt > asyncOperation.createdAt);
+  t.true(fetchedRecord.updatedAt > asyncOperation.updatedAt);
+
+  const esRecord = await t.context.esAsyncOperationsClient.get(asyncOperation.id);
+
+  t.is(asyncOperationPgRecord.created_at.getTime(), record.createdAt);
+  t.is(asyncOperationPgRecord.updated_at.getTime(), record.updatedAt);
+  t.is(asyncOperationPgRecord.created_at.getTime(), esRecord.createdAt);
+  t.is(asyncOperationPgRecord.updated_at.getTime(), esRecord.updatedAt);
+});
+
+test('POST returns a 409 error if the async operation already exists in DynamoDB', async (t) => {
+  const { asyncOperationModel, jwtAuthToken } = t.context;
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  await asyncOperationModel.create(asyncOperation);
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .send(asyncOperation)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(409);
+  const { message } = response.body;
+  t.is(message, (`A DynamoDb record already exists for async operation ID ${asyncOperation.id}`));
+});
+
+test.serial('POST returns a 500 response if record creation throws unexpected error', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const stub = sinon.stub(AsyncOperationModel.prototype, 'create')
+    .callsFake(() => {
+      throw new Error('unexpected error');
+    });
+
+  const asyncOperation = {
+    id: uuidv4(),
+    status: 'RUNNING',
+    taskArn: randomString(),
+    description: 'Some async run',
+    operationType: 'Bulk Granules',
+    output: JSON.stringify({ age: 59 }),
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  };
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .send(asyncOperation)
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .expect(500);
+
+  t.is(response.status, 500);
+  t.teardown(() => {
+    stub.restore();
+  });
+});
+
+test('POST returns a 400 response if invalid record is provided', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const asyncOperation = { invalid: 'value' };
+
+  const response = await request(app)
+    .post('/asyncOperations')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(asyncOperation)
+    .expect(400);
+  t.is(response.status, 400);
+});
+
+test('POST returns a 404 if the requested path does not exist', async (t) => {
+  const { jwtAuthToken } = t.context;
+  const asyncOperation = {
```

Comment:
FWIW, I think we can use the `fakeAsyncOperationFactory` from `api/lib/testUtils` rather than repeating this in efvery test. But `fakeAsyncOperationFactory` should be updated to use a UUID for the `id`

Reply:
[`16e0540` (#2294)](https://github.com/nasa/cumulus/pull/2294/commits/16e0540d6595599dab2f7929b4a98ec0940a6bf4)